src/cljam/io/vcf/util/normalize.clj

```diff
@@ -0,0 +1,173 @@
+      (Character/toUpperCase ^char y)))
+  ([x y & z]
+   (and (char-equals-ignore-case? x y)
+        (apply char-equals-ignore-case? y z))))
+
+(defn- trim-right
+  "Trims all duplicated allele sequences from right. If reached the start,
+  reference sequences of length `window` will be read from `seq-reader` and
+  be prepended to the alleles."
+  [seq-reader {:keys [chr ^long pos ref alt]
+               {:keys [END]} :info :as v} ^long window]
+  (let [alleles (cons ref alt)
+        min-end (long (dec (+ pos (apply min (map count alleles)))))
+        regs (regions-before chr pos window)
+        ref-seqs (sequence ;; unchunk
+                  (mapcat
+                   (fn [r]
+                     (-> seq-reader
+                         (io-seq/read-sequence r {:mask? false})
+                         cstr/reverse)))
+                  regs)
+        allele-seqs (map #(concat (cstr/reverse %) ref-seqs) alleles)
+        matched-length (->> allele-seqs
+                            (apply map char-equals-ignore-case?)
+                            (take-while true?)
+                            count
+                            long
+                            (Math/min (dec min-end)))]
+    (if (pos? matched-length)
+      (let [pos' (or (some (fn [{:keys [start end]}]
+                             (when (<= start (- min-end matched-length) end)
+                               start)) regs)
+                     pos)
+            [ref' & alt'] (map
+                           #(->> %2
+                                 (take (inc (- (dec (+ pos (count %1))) pos')))
+                                 (drop matched-length)
+                                 reverse
+                                 cstr/join)
+                           alleles
+                           allele-seqs)]
+        (-> (if (integer? END)
+              (assoc-in v [:info :END] (dec (+ pos' (count ref'))))
+              v)
+            (assoc :pos pos' :ref ref' :alt alt')))
+      v)))
+
+(defn- trim-left
+  "Trims all duplicated allele sequences from left."
+  [{:keys [ref alt] :as v}]
+  (let [alleles (cons ref alt)
+        n-trim-left (->> alleles
+                         (apply map char-equals-ignore-case?)
+                         (take-while true?)
+                         count)
+        min-len (long (apply min (map count alleles)))
+        n-trim-left (min (dec min-len) n-trim-left)]
+    (if (pos? n-trim-left)
+      (-> v
+          (update :pos + n-trim-left)
+          (update :ref subs n-trim-left)
+          (assoc :alt (map #(subs % n-trim-left) alt)))
+      v)))
+
+(def ^:private ^:const nucleotides-regexp #"(?i)[ATGCN]+")
+
+(defn realign
+  "Left-align and trim REF/ALT alleles."
+  ([seq-reader variant]
+   (realign seq-reader variant {}))
+  ([seq-reader
+    {:keys [ref alt] :as variant}
+    {:keys [window] :or {window 100}}]
+   (cond
+     (and
+      ref
+      (seq alt)
+      (re-matches nucleotides-regexp ref)
```

Comment:
I think the check against `ref` is a little bit redundant because `vcf-util/inspect-allele` can also play the role.
https://github.com/chrovis/cljam/blob/55353b7bee71cda5313655d2c91970094a35c728/src/cljam/io/vcf/util.clj#L358-L373

Reply:
That's a good point. Thanks! Fixed in 9e05452